File: longest-substring-no-duplicates/python/longest-substring.py
def lengthOfLongestSubstring(self, s):
    """
    :type s: str
    :rtype: int
    """
    if s == "":
        return 0
        
    maxLength = 1
    for i in range(len(s)):
        charDict = {s[i]: True}
        currentLength = 1
        for j in range(i + 1, len(s)):
            if s[j] in charDict:
                break
            currentLength += 1
            charDict[s[j]] = True
            maxLength = max(maxLength, currentLength)
                
    return maxLength

File: longest-substring-no-duplicates/java/longest-substring.java
public int lengthOfLongestSubstring(String s) {
    if(s.isEmpty()) return 0;
    Integer maxLength = 1;
        
    for(Integer i = 0; i < s.length(); i++){
        Integer currentLength = 1;
        HashMap<Character, Boolean> charsMap = new HashMap<>();
        charsMap.put(s.charAt(i), true);
        for(Integer j = i + 1; j < s.length(); j++){
            if(charsMap.containsKey(s.charAt(j))){
                break;
            }
            currentLength++;
            charsMap.put(s.charAt(j), true);
            maxLength = Math.max(maxLength, currentLength);
        }
    }
    return maxLength;
}